src/ncp-spinel/SpinelNCPControlInterface.cpp

```diff
@@ -129,6 +129,11 @@ SpinelNCPControlInterface::attach(CallbackWithStatus cb)
 	);
 }
 
+static bool check_ncp_finished_initalization(SpinelNCPInstance *instance)
+{
+	return !ncp_state_is_initializing(instance->get_ncp_state()) && !instance->is_initializing_ncp();
```

Comment:
Will there be such case that the final_check is called even before the ncp instance sets its state to initializing and starts to reset the device? Shall we also check ncp is not in a "working" state?